src/parquet/column_writer.cc

```diff
@@ -33,6 +33,8 @@
 
 namespace parquet {
 
+static constexpr int MAX_STATS_SIZE = 4096;  // limit stats to 4k
```

Comment:
Is this defined by the Parquet standard or arbitrarily chosen?

Reply:
I followed parquet-mr.
https://github.com/apache/parquet-mr/blob/0d55abd05b0e5027c18e60d1ac3b22998dd00951/parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java#L88
I don't see this specified in the spec. We should probably add it there. I will open a JIRA to do this.

Reply:
The alternative would be to use this as sensible default and add it as an option to `WriterProperties`.

Reply:
I added it to `ColumnProperties` since it gives more flexibility.